src/scriptengine/ScriptEngine.hpp

```diff
@@ -51,6 +51,13 @@ class ScriptEngine
   // execute string without expecting a return value
   virtual void exec(std::string_view sv) = 0;
 
+  // TODO: this is totally the wrong place to put it, but I'm trying to see if it works
+  virtual void setupEmbeddedGems(const std::vector<openstudio::path>& includeDirs, const std::vector<openstudio::path>& gemPathDirs,
+                                 const openstudio::path& gemHomeDir, const openstudio::path& bundleGemFilePath,
+                                 const openstudio::path& bundleGemDirPath, const std::string& bundleWithoutGroups){};
+
+  virtual void setupPythonPath(const std::vector<openstudio::path>& includeDirs, const openstudio::path& pythonHomeDir){};
```

Comment:
I need a way to pass from cli/main.cpp to the Ruby/Python Engines, but all I have is a ScriptEngineInstance. So I add a virtual function, which is empty in the base class, and only one of these two is overriden in the respective Ruby/Python Engine. I'm actually thinking that TODO was too harsh, I can't think of another way.

@kbenne Maybe another area for us to discuss.